packages/pluggableWidgets/notifications-native/src/Notifications.tsx

```diff
@@ -38,9 +39,14 @@ interface Notification {
 export function Notifications(props: NotificationsProps<undefined>): null {
     const listeners = useRef<Array<() => void>>([]);
     const [loadNotifications, setLoadNotifications] = useState(false);
+    const [knownIds] = useState(new Set<string>());
```

Comment:
Normally, you use the setter that you receive here (`const [knownIds, setKnownIds] = ...`) to update state variables. I'm actually not sure if you should do that here as well

Reply:
I too find this peculiar. If you don't want to use the "state setter", causing a re-render, you could use a `ref` (`useRef`), which maintains it's value throughout the components lifecycle. This would then mitigate needing to add the ref to any dependency array (e.g. line 79).

Using a piece of state from `useState` without its setter could lead to very weird scenarios; for example, IIUC, your `useCallback` could reference a state of `knownIds` that you don't expect.

Reply:
Thanks Jordan for your help, the latest commit contain the resolution using useRef